src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaBaseClassTests.cs

```diff
@@ -200,16 +189,32 @@ public static void CallsSignDataCore(SlhDsaAlgorithm algorithm)
             Array.Fill(signature, (byte)42);
             slhDsa.SignData(testData, signature.AsSpan(1, signatureSize), testContext);
 
-            Assert.Equal(42, signature[0]);
-            Assert.Equal(42, signature[^1]);
-            Assert.All(signature.Skip(1).SkipLast(1), b => Assert.Equal(1, b));
+            AssertExpectedFill(signature, nameof(SlhDsa.SignData), callNumber: 1);
 
             Array.Fill(signature, (byte)42);
-            slhDsa.SignData(testData, signature.AsSpan(1, signatureSize), testContext); // Extra byte should be ignored
+            slhDsa.SignData(testData, signature.AsSpan(1, signatureSize + 1), testContext); // Extra byte should be ignored
 
-            Assert.Equal(42, signature[0]);
-            Assert.Equal(42, signature[^1]);
-            Assert.All(signature.Skip(1).SkipLast(1), b => Assert.Equal(1, b));
+            AssertExpectedFill(signature, nameof(SlhDsa.SignData), callNumber: 2);
+        }
+
+        private static void AssertExpectedFill(ReadOnlySpan<byte> source, string functionBeingTested, int callNumber)
+        {
+            if (source[0] != 42)
+            {
+                throw new XunitException($"Call #{callNumber} to {functionBeingTested} overwrote data before destination[0].{Environment.NewLine}Expected: 42{Environment.NewLine}Actual: {source[0]}");
+            }
+
+            if (source[^1] != 42)
+            {
+                throw new XunitException($"Call #{callNumber} to {functionBeingTested} overwrote data beyond the indicated length.{Environment.NewLine}Expected: 42{Environment.NewLine}Actual: {source[^1]}");
+            }
+
+            int idx = source.Slice(1, source.Length - 2).IndexOfAnyExcept((byte)1);
```

Comment:
There's already an assertion helper for "filled with". We should use that, or improve it as needed.

https://github.com/dotnet/runtime/blob/472c3b042ba8426daab014d61ff2de26055bec3f/src/libraries/Common/tests/System/Security/Cryptography/AesGcmTests.cs#L487

Reply:
Updated to use `AssertExtensions.FilledWith`. The pattern to check leading and trailing padding is used in other places as well so I think it's good to keep following that.